tools/build-tools/src/test/typeValidator/interfaceValidator.spec.ts

```diff
@@ -0,0 +1,391 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import os from "os";
+import { Project, SourceFile } from "ts-morph"
+import { createSpecificValidator } from "./../../typeValidator/packageValidator";
+import { TypeData } from "../../typeValidator/typeData";
+import { BreakingIncrement, enableLogging } from "./../../typeValidator/validatorUtils"
+
+describe("Interface", () => {
+    enableLogging(true);
+    let project: Project;
+    let pkgDir: string = os.tmpdir();
+    before(() => {
+        project = new Project({
+            skipFileDependencyResolution: true,
+        });
+        assert(project !== undefined);
+    });
+
+    function getTypeDataForSource(sourceFile: SourceFile): TypeData {
+        let typeData: TypeData;
+        for (const declarations of sourceFile.getExportedDeclarations().values()) {
+            typeData = { kind: "unknown", name: "typeName", node: declarations[0] };
+            break;
+        }
+        return typeData!;
+    }
+
+    function checkIncrement(
+        oldSource: string,
+        newSource: string,
+    ): BreakingIncrement {
+        const oldSourceFile = project.createSourceFile(`${pkgDir}/src/interfaceOld.ts`, oldSource, { overwrite: true });
+        const oldTypeData = getTypeDataForSource(oldSourceFile);
+        const newSourceFile = project.createSourceFile(`${pkgDir}/src/interfaceNew.ts`, newSource, { overwrite: true });
+        const newTypeData = getTypeDataForSource(newSourceFile);
+
+        const validator = createSpecificValidator(
+            project.getTypeChecker(),
+            oldTypeData.node,
+            project.getTypeChecker(),
+            newTypeData.node,
+        );
+
+        return validator.validate(project, pkgDir);
+    }
```

Comment:
Should this be shared between the class and interface tests?

Reply:
yeah probably.  there's a lot of refactoring that can be done here i think that just hasn't been done yet :/